src/Sentry/Internal/SentryScopeManager.cs

```diff
@@ -91,40 +87,52 @@ public IDisposable PushScope<TState>(TState? state)
         ScopeAndClientStack = newScopeAndClientStack;
         return scopeSnapshot;
     }
+    public void RestoreScope(Scope savedScope)
+    {
+        var currentScopeAndClientStack = ScopeAndClientStack;
```

Comment:
This method isn't safe when IsGlobalMode=true. It doesn't seem like it would be called the way things are right now, but best to guard it in here to avoid surprises

Similar to this: https://github.com/getsentry/sentry-dotnet/blob/407dc28dbd7f1c29059fc509a886b114a2c75b0b/src/Sentry/Internal/SentryScopeManager.cs#L55-L59